### PR TITLE
ARROW-8356: [Developer] Support * wildcards with "crossbow submit" via GitHub actions

### PR DESCRIPTION
**Pull request by kszucs**

### CODE REVIEW

**Comment by kszucs** [comment]
@github-actions crossbow submit conda-* --dry-run

**Comment by kszucs** [comment]
@github-actions crossbow submit --dry-run conda-linux-*

**Comment by kszucs** [comment]
Ohh, forget that the --dry-run change must land in the master before taking effect.

File: dev/tasks/crossbow.py [hunk 1]
```diff
@@ -953,16 +978,26 @@ def from_config(cls, config, target, task_whitelist=None,
...
             raise click.ClickException(msg)
 
-        requested_tasks = [list(config_groups[name])
-                           for name in group_whitelist]
-        requested_tasks = set(sum(requested_tasks, task_whitelist))
+        # merge the tasks defined in the selected groups
+        task_patterns = [list(config_groups[name]) for name in group_whitelist]
+        task_patterns = set(sum(task_patterns, task_whitelist))
+
+        # treat the task names as glob patters to select tasks more easily
```
Comment (bkietz) [inline dev/tasks/crossbow.py:985]:
```suggestion
        # treat the task names as glob patterns to select tasks more easily
```

File: dev/tasks/crossbow.py [hunk 2]
```diff
@@ -936,15 +937,39 @@ def date(self):
...
+        """
+        Intantiate a job from based on a config.
+
+        Parameters
+        ----------
+        config : dict
+            Deserialized content of tasks.yml
+        target : Target
+            Describes target repository and revision the builds run against.
+        tasks : Optiona[List[str]], default None
```
Comment (bkietz) [inline dev/tasks/crossbow.py:950]:
```suggestion
        tasks : Optional[List[str]], default None
```

File: dev/tasks/crossbow.py [hunk 3]
```diff
@@ -936,15 +937,39 @@ def date(self):
...
+            List of glob patterns for matching task names.
+        groups : tasks : Optiona[List[str]], default None
```
Comment (bkietz) [inline dev/tasks/crossbow.py:952]:
```suggestion
        groups : tasks : Optional[List[str]], default None
```

**Review by bkietz — CHANGES_REQUESTED** [review]
Some typos, otherwise this seems fine

**Comment by kszucs** [comment]
Thanks @bkietz! Updated.

**Review by bkietz — APPROVED** [review]

**Comment by wesm** [comment]
The reduction in task configuration is great

**Comment by nealrichardson** [comment]
I don't think this was tested enough. I'm looking into some fixes on #6868 but am hitting some errors and unexpected behavior (importantly, among others, the nightly group doesn't run)

**Comment by wesm** [comment]
Bummer, the absence of unit tests for Crossbow is a known liability

**Pull request closed without merging**